File: 697 - Degree of an Array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findShortestSubArray = function (nums) {
    let counts = new Map();
    for (let i = 0; i < nums.length; i++) {
        if (counts.has(nums[i])) {
            counts.get(nums[i])[0]++;
            counts.get(nums[i])[2] = i;
        } else {
            counts.set(nums[i], [1, i, i]);
        }
    }
    let maxcount = 0, minlen = 0;
    for (let i of counts.entries()) {
        let count = i[1][0],
            left = i[1][1],
            right = i[1][2];
        if (maxcount < count) {
            maxcount = count;
            minlen = right - left + 1;
        } else if (maxcount == count) {
            minlen = Math.min(minlen, right - left + 1);
        }

    }
    return minlen;
};


let a = findShortestSubArray([1, 2, 3, 4, 5, 4, 2]);
console.log(a);

File: 697 - Degree of an Array.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int findShortestSubArray(int[] nums) {
        HashMap<Integer, Integer[]> counts = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (counts.containsKey(nums[i])) {
                counts.get(nums[i])[0]++;
                counts.get(nums[i])[2] = i;
            } else {
                counts.put(nums[i], new Integer[] {1, i, i});
            }
        }
        int maxcount = 0, minlen = 0;
        for (Map.Entry<Integer, Integer[]> entry : counts.entrySet()) {
            int count = entry.getValue()[0], 
                left = entry.getValue()[1],
                right =  entry.getValue()[2];

            if (count > maxcount) {
                maxcount = count;
                minlen = right - left + 1;
            } else if (count == maxcount) {
                minlen = Math.min(minlen, right - left + 1);
            }
        }
        return minlen;

    }
}